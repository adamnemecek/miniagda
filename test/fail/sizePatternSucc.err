MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "sizePatternSucc.ma" ---
--- scope checking ---
--- type checking ---
type  SNat : + Size -> Set
term  zero : .[i : Size] -> SNat $i
term  succ : .[i : Size] -> ^ SNat i -> SNat $i
type  SNat : + Size -> Set
type  Empty : Set
type  Empty : Set
term  bad : .[i : Size] -> SNat i -> Empty
error during typechecking:
bad
/// clause 2
/// pattern zero $i
/// pattern $i
/// successor pattern only allowed in cofun
