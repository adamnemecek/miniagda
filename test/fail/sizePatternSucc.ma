
sized data SNat : Size -> Set
{
zero : (i : Size) -> SNat ($ i);
succ : (i : Size) -> SNat i -> SNat ($ i)
}

data Empty : Set
{
}

-- ($ i) appearing as a size pattern

fun bad : (i : Size) -> SNat i -> Empty
{
bad .($ i)     (succ i x)   = bad _ x;
bad .($ ($ i)) (zero ($ i)) = bad _ (zero _);
}
