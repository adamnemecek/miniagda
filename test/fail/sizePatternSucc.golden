MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "sizePatternSucc.ma" ---
--- scope checking ---
--- type checking ---
type  SNat : + Size -> Set
term  SNat.zero : .[s!ze : Size] -> .[i < s!ze] -> SNat s!ze
term  SNat.zero : .[i : Size] -> < SNat.zero i : SNat $i >
term  SNat.succ : .[s!ze : Size] -> .[i < s!ze] -> ^ SNat i -> SNat s!ze
term  SNat.succ : .[i : Size] -> ^(y1 : SNat i) -> < SNat.succ i y1 : SNat $i >
type  Empty : Set
term  bad : .[i : Size] -> SNat i -> Empty
error during typechecking:
bad
/// clause 2
/// pattern zero $i
/// pattern $i
/// successor pattern only allowed in cofun
